File: python/02_cryptographic_failures.py
"""A02 - Cryptographic Failures. VULN: MD5 password + hardcoded key."""
import hashlib

# VULN: A02 - weak hash, no salt
SECRET_KEY = "hardcoded-dev-key-change-me"  # VULN: hardcoded secret

def hash_password(password: str) -> str:
    return hashlib.md5(password.encode()).hexdigest()

File: c/02_cryptographic_failures.c
/* A02 - Cryptographic Failures. VULN: custom XOR "encryption". */
#include <string.h>

static const char *KEY = "secret"; /* VULN: A02 hardcoded */

void xor_crypt(char *buf, size_t len) {
  for (size_t i = 0; i < len; i++)
    buf[i] ^= KEY[i % strlen(KEY)];
}
